### PR TITLE
JS: More robust CommonJS/ES2015 detection logic for extractor

### PR DESCRIPTION
**Pull request by RasmusWL**
review commit-by-commit recommended.

These changes caused a few unrelated trap files to change as well, I think these are all for programs with syntax-errors, but would certainly like you to take a look at these @asgerf (see 60944a9)

I think the change-note for this need to go into the internal repo for the CLI.

### CODE REVIEW

File: javascript/extractor/tests/flow/output/trap/anonFunctionWithoutParens.js.trap [hunk 1]
```diff
@@ -72,20 +72,14 @@ locations_default(#20025,#10000,1,40,1,39)
...
-successor(#20027,#20029)
+#20026=*
+entry_cfg_node(#20026,#20001)
+#20027=@"loc,{#10000},1,1,1,0"
+locations_default(#20027,#10000,1,1,1,0)
+hasLocation(#20026,#20027)
+#20028=*
+exit_cfg_node(#20028,#20001)
+hasLocation(#20028,#20025)
+successor(#20026,#20028)
```
Comment (asgerf) [inline javascript/extractor/tests/flow/output/trap/anonFunctionWithoutParens.js.trap:83]: These files are written in a JS dialect called Flow (which is why they are in a folder called `flow`). We tolerate the presence of Flow type annotations and type alias declarations, but we don't extract them to the AST, which is why the `export type` declarations aren't seen by the new detector.

I'll think about the best course of action and follow up.

**Review by asgerf — COMMENTED** [review]
Great work! I've started a DCA run.

> I think the change-note for this need to go into the internal repo for the CLI.

We usually put extractor change notes here next to QL-related change notes.

File: javascript/extractor/tests/flow/output/trap/anonFunctionWithoutParens.js.trap (hunk 1 above)
Comment (RasmusWL) [reply javascript/extractor/tests/flow/output/trap/anonFunctionWithoutParens.js.trap:83]: Aha, that all makes more sense now :+1: 

> I'll think about the best course of action and follow up.

Sounds good :+1:

**Comment by RasmusWL** [comment]
> We usually put extractor change notes here next to QL-related change notes.

Cool, I have created one now :+1:

**Review by asgerf — APPROVED** [review]
LGTM 👍

**Pull request merged**